src/btree/btree.rs

```diff
@@ -0,0 +1,509 @@
+// Copyright 2022 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+//! BTree structure.
+
+use super::*;
+use crate::table::key::TableKeyQuery;
+use crate::log::{LogWriter, LogQuery};
+use crate::column::Column;
+use crate::error::{Error, Result};
+use parking_lot::RwLock;
+
+/// In memory local btree overlay.
+
+pub struct BTree {
+	pub(super) depth: u32,
+	pub(super) root: Box<Node>,
+	pub(super) root_index: Option<u64>,
+	pub(super) removed_children: RemovedChildren,
+	pub(super) record_id: u64,
+}
+
+pub struct BTreeIterator<'a> {
+	db: &'a crate::db::DbInner,
```

Comment:
I remember trying to switch to it, but It also requires an access to the commit overlay.
Maybe I should move this struct to src/btree/mod.rs or src/db.rs since it is really external api.

Reply:
I think it is still better to keep a reference to commit overlay instead of the whole database. It also help moving iteration logic to the btree module. I've tried to do this in this commit: https://github.com/paritytech/parity-db/commit/a70f71ee3ba8f9f5f6d6bd7b716e2c31e0b8c043
Maybe it would be better to move this to its own `iter` module indeed.